web_api.md

```diff
@@ -0,0 +1,318 @@
+```
+
+There are three main sections in the configuration file.
+
+1. `message_queue`: it contains the `address` and `port`, where the queue was deployed, or will be deployed with the
+   specified `credentials`. If the `ssh` property is presented, the Processing Server will try to connect to
+   the `address` via `ssh` with provided `username` and `password` and deploy [RabbitMQ](https://www.rabbitmq.com/) at
+   the specified `port`. The remote machine must have [Docker](https://www.docker.com/) installed since the deployment
+   is done via Docker. Make sure that the provided `username` has enough rights to run Docker commands. In case
+   the `ssh` property is not presented, the Processing Server assumes that RabbitMQ was already deployed and just uses
+   it.
+2. `mongo_db`: this section also contains the `address` and `port`, where the [MongoDB](https://www.mongodb.com/) is
+   running, or will run. If `credentials` is presented, it will be used when deploying and connecting to the database.
+   The `ssh` section behaves exactly the same as described in the `message_queue` section above.
+3. `hosts`: this section contains a list of hosts, usually virtual machines, where Processing Workers should be
+   deployed. To be able to connect to a host, an `address` and `username` are required, then comes either `password`
+   or `path_to_privkey` (path to a private key). All Processing Workers, which will be deployed, must be declared under
+   the `deploy_processors` property. In case `type` is `docker`, make sure that [Docker](https://www.docker.com/) is
+   installed in the target machine and the provided `username` has enough rights to execute Docker commands.
+
+Among three sections, only the `message_queue` is required. However, if `hosts` is present, `mongo_db` must be there
+as well. For more information, please check the [configuration file schema](web_api/config.schema.yml).
+
+### Processing Worker
+
+There is normally no need to start (or stop) a Processing Worker manually, since it can be managed by a Processing
+Server via a [configuration file](#processing-server). However, if it is necessary to do so, there are two ways to start
+a Processing Worker:
+
+```shell
+# 1. Use ocrd CLI bundled with OCR-D/core
+$ ocrd processing-worker <processor-name> --queue=<queue-address> --database=<database-address>
+
+# 2. Use processor name
+<processor-name> processing-worker --queue=<queue-address> --database=<database-address>
+```
+
+* `--queue`: a [Rabbit MQ connection string](https://www.rabbitmq.com/uri-spec.html) to a running instance.
+* `--database`: a [MongoDB connection string](https://www.mongodb.com/docs/manual/reference/connection-string/) to a
+  running instance.
+
+### Process Queue
+
+By using a queuing system, specifically [RabbitMQ](https://www.rabbitmq.com/), the reliability and flexibility of the
+whole system are greatly improved. In our implementation, manual acknowledgement mode is used. This means, when a
```

Comment:
...improved over **what**?

```suggestion
Processing Server are greatly improved over a system directly coupling the workflow engine and distributed processor instances.

 In our implementation of the Process Queue, manual acknowledgement mode is used. This means, when a
```

Would that be correct?

Reply:
It is improved compared to other approaches, like distributed REST calls or local CLI calls.